sktime/forecasting/compose/_pipeline.py

```diff
@@ -1583,10 +1583,14 @@ def _get_X_for_fcX(self, X):
             return None
 
         if ixx == "complement":
-            X_for_fcX = X.drop(columns=self.columns)
-            if X_for_fcX.shape[1] < 1:
+            if X is None:
                 return None
-            return X_for_fcX
+            all_columns = X.columns
+            if self.columns is not None:
+                complemented_columns = list(set(all_columns) - set(self.columns))
```

Comment:
To keep the original column order, we can try list comprehension. This is probably causing the CI failures as @fkiraly pointed out earlier.

```suggestion
                complemented_columns = [column for column is all_columns if column not in self.columns]
```

(This snippet is not tested and likely breaks 88 line limit.)

Reply:
sure, will use pre-commit to check all lint failures